store/database.go

```diff
@@ -92,6 +92,19 @@ func (s *Store) FindDatabase(ctx context.Context, find *api.DatabaseFind) ([]*ap
 		}
 		databaseList = append(databaseList, database)
 	}
+
+	// If no specified instance, filter out databases belonging to archived instances.
+	if find.InstanceID == nil {
+		var filteredList []*api.Database
+		for _, database := range databaseList {
+			if i := database.Instance; i != nil && i.RowStatus == api.Archived {
```

Comment:
Shall we skip the ones that has database.instance == nil?

```
if database.Instance == nil {
  continue
}
if database.Instance.RowStatus == api.Archived {
  continue
}
...
```

Reply:
actually all databases have instance, this is to cheat linter

Reply:
How about i == nil || i.RowStatus == api.Archived for logical completeness?